templates/modelExpt/run_modelComp_expt.m

```diff
@@ -0,0 +1,170 @@
+%   speech samples and allow the experimenter to evaluate different LPCs on
+%   that dataset. For simplicity here, LPC is just set arbitrarily.
+if strcmp(expt.gender, 'male')
+    expt.audapterParams.nLPC = 17;
+else
+    expt.audapterParams.nLPC = 15;
+end
+
+%perturbation amount, in mels
+shiftMag = 125;
+
+% timing
+expt.timing.stimdur = 1.9;         % time stim is on screen, in seconds
+expt.timing.interstimdur = .75;    % minimum time between stims, in seconds
+expt.timing.interstimjitter = .75; % maximum extra time between stims (jitter)
+expt.timing.visualfbdur = 0.75; 
+
+%set up duration feedback parameters
+expt.durcalc.min_dur = .4; % set_exptDefaults normally sets to .25
+expt.durcalc.max_dur = .65;
+expt.durcalc.ons_thresh = 0.15;
+expt.durcalc.offs_thresh = 0.4;
+
+%% Set up duration practice
+exptDur = expt;
+exptDur.session = 'dur'; 
+exptDur.dataPath = fullfile(expt.dataPath,exptDur.session);
+if ~exist(exptDur.dataPath, 'dir')
+    mkdir(exptDur.dataPath)
+end
+exptDur.ntrials = 6;
+
+exptDur.shiftMags   = zeros(1,exptDur.ntrials);
+exptDur.shiftAngles = zeros(1,exptDur.ntrials);
+
+exptDur.words = {'head' 'dead' 'Ted'};
+exptDur.allWords = mod(0:exptDur.ntrials-1, numel(exptDur.words)) + 1;
+exptDur.listWords = exptDur.words(exptDur.allWords);
+
+exptDur.conds = {'noShift'};
+exptDur.allConds = ones(1, exptDur.ntrials);
+exptDur.listConds = exptDur.conds(exptDur.allConds);
+
+% set up duration feedback parameters
+exptDur.bDurFB = ones(1,exptDur.ntrials); % 1 x ntrials vector
+
+exptDur = set_exptDefaults(exptDur);
+
+exptfile = fullfile(exptDur.dataPath,'expt.mat');
+dur.expt = exptDur;
+save(exptfile, '-struct', 'dur')
+fprintf('Saved exptDur file: %s.\n',exptfile);
+
+%% Set up compensation experiment
+% stimuli and vowel list
+expt.words = {'head' 'dead' 'Ted'}; 
+nwords = length(expt.words);
+[vowels,~,ivowels] = unique(txt2arpabet(expt.words));
+if length(vowels) == length(ivowels), vowels = vowels(ivowels); end
+expt = set_missingField(expt,'vowels',vowels);
+
+expt.shiftDirs = {0 -1 1};
+expt.shiftNames = {'noShift' 'shiftDown' 'shiftUp'};
+expt.shiftMag = 125;
+expt.conds = {'noShift' 'shiftDown' 'shiftUp'};
+
+if bTestMode
+    expt.nblocks = 1;
+else
+    expt.nblocks = 6;
+end
+
+expt.ntrials_per_block = nwords*(2+3); %aka 15 trials per block.
+
+% Number of trials that don't receive perturbation
+if bTestMode 
+    nBaseline = 1;
+else 
+    nBaseline = 15;
+end
+
+% Pseudorandomize stimuli like this: In each 15-trial block, each word gets
+% an equal number of trials. Within a word, there are 2 perturbation
+% trials, and 3 unperturbed trials. Within a word's 2 perturbation trials,
+% there's 1 UP and 1 DOWN perturbation. No perturbation trials are adjacent, even across blocks.
+expt = randomize_stimuli(expt,1,3,nBaseline);
+
+% Set up ntrials, coherence, and direction
+expt.ntrials = expt.nblocks * expt.ntrials_per_block;
+
+% Set up breaks, expt.ntrials must be divisible by break frequency
```

Comment:
I don't think ntrials needs to be divisible by break frequency, does it?
(Also, if you set expt.breakFrequency, you don't need to set expt.breakTrials; set_exptDefaults will do it for you)

Reply:
correct, it doesn't. removed expt.breakTrials set and replaced w/ comment